cql3/statements/select_statement.cc

```diff
@@ -266,6 +266,12 @@ uint64_t select_statement::do_get_limit(const query_options& options, ::shared_p
   });
 }
 
+db::timeout_clock::duration select_statement::get_timeout(const service::client_state& state, const query_options& options) const {
+    return state.get_session_params().latency_limit_for_reads.value_or(
+        options.get_timeout_config().*get_timeout_config_selector()
+    );
```

Comment:
Suggest to populate the session timeout from the global timeout if not specified. This way, the runtime path is if-less.

Reply:
(in fact we can just populate query_options::_timeout_config, then no changes are needed here)

Reply:
That's a great idea, we already have the information in place when query_options are created. Will do.

Reply:
Actually, `query_options` only keeps a reference to the global server config, which can't simply be modified. So it's a question of a branch in the code vs passing the whole timeout config for each query options. I think that the timeout config is heavy enough to avoid copying it extensively (7 `std::duration`s now, can grow).

Reply:
We can make it refer to the session instead of the global config (and seed the session config from the global config).

Reply:
Yeah, but then we'll also need to verify if these timeouts are reloadable online from config. If they are, we'd need to propagate changes to new sessions too, which makes it a little more complicated. But maybe these configs are only reloadable upon restart, I haven't checked.

Reply:
*existing sessions, not new ones

Reply:
Even if we add reload, we can make it re-propagate the timeouts to sessions with no timeout overrides.

Reply:
Right, that's even better. A single copy of the timeout config will be added for each connection, but in return, the timeout config reference for each executed statement will not be needed anymore, so it's usually a net win for memory usage (unless somebody uses a new connection for each statement, in which case they're doomed anyway).

Reply:
In fact we're all doomed.